clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaBasedCommandSender.java

```diff
@@ -274,14 +276,38 @@ void setCorrelationIdSupplier(final Supplier<String> correlationIdSupplier) {
                         .map(ids -> ids.remove(message.getCorrelationId()))
                         .ifPresentOrElse(
                                 expiringCommandPromise -> expiringCommandPromise
-                                        .tryCompleteAndCancelTimer(Future.succeededFuture(message)),
+                                        .tryCompleteAndCancelTimer(mapResponseResult(message)),
                                 () -> LOGGER.trace(
                                         "ignoring received command response [correlation-id: {}] as there is no matching correlation id found",
                                         message.getCorrelationId()));
             }
         };
     }
 
+    /**
+     * Maps the status according to
+     * {@link CommandSender#sendCommand(String, String, String, String, Buffer, String, Map, SpanContext)}.
+     *
+     * @param message The received command response.
+     * @return The outcome: a succeeded future if the message contains a 2xx status and a failed future otherwise.
+     */
+    private Future<DownstreamMessage<KafkaMessageContext>> mapResponseResult(
+            final DownstreamMessage<KafkaMessageContext> message) {
+
+        final int status = Optional.ofNullable(message.getStatus()).orElseGet(() -> {
+            LOGGER.trace(
```

Comment:
I think, this can be logged on DEBUG level (at least) because such a case would most likely be a bug in the protocol adapter that forwarded the command response.

Reply:
I wanted it to match the AMQP pendant. But I agree, this violates the C&C API. If it indicates a bug in the adapter, we should consider an even higher level. What about WARN? And then change it in `ProtonBasedRequestResponseCommandClient` as well.

Reply:
Yes, good.